src/forms/TextInput.jsx

```diff
@@ -77,6 +77,10 @@ export const TextInput = (props) => {
 		other.value = value;
 	}
 
+	// Character limits should be a "soft" limit.
+	// Avoid passing maxLength as an HTML attribute
+	if (maxLength) delete other.maxLength;
```

Comment:
do we want this to be a universal thing? i would assume making it a soft limit would be an option

Reply:
We do in fact want to make this universal. There are better reasons to use a soft limit than a hard limit, and it's nice to codify this product decision in our components.